### ISSUE TITLE
DDG should support client root span

### ISSUE THREAD

**Issue by cykl** [jaegertracing/jaeger-ui#546]
## Requirement - what kind of business use case are you trying to solve?

I want to display the DDG of a trace started by a batch job. 

## Problem - what in Jaeger blocks you from solving the requirement?

Root spans of traces started by batch job are often client spans, ie. span.kind = client, but current DDG implementation requires a server root span. If this condition is not fulfilled DDG is not shown and an error message is displayed: _There are no dependencies. No traces were found that contain client and any other service where span.kind is ‘server’._

I believe client root span ares valid use cases and should be supported.

## Minimal test case

Following code produces a trace matching my issue:

```java
import io.jaegertracing.internal.JaegerTracer;
import io.jaegertracing.internal.reporters.RemoteReporter;
import io.jaegertracing.internal.samplers.ConstSampler;
import io.jaegertracing.thrift.internal.senders.UdpSender;
import io.opentracing.Tracer;
import io.opentracing.tag.Tags;

public class ClientRootSpan {

    public static void main( String[ ] args ) throws InterruptedException {
        try (var clientTracer = createTracer( "client" )) {
            var clientSpan = clientTracer.buildSpan( "GET" )
                    .withTag( Tags.SPAN_KIND, "client" )
                    .withTag( Tags.HTTP_METHOD, "GET" )
                    .start( );

            Thread.sleep( 100 );

            try (var serverTracer = createTracer( "server" )) {
                var serverSpan = serverTracer.buildSpan( "GET" )
                        .asChildOf( clientSpan )
                        .withTag( Tags.SPAN_KIND, "server" )
                        .withTag( Tags.HTTP_METHOD, "GET" )
                        .start( );

                Thread.sleep( 50 );
                serverSpan.finish( );
            }

            clientSpan.finish( );
        }
    }

    static Tracer createTracer( String serviceName ) {
        return new JaegerTracer.Builder( serviceName )
                .withSampler( new ConstSampler( true ) )
                .withReporter( new RemoteReporter.Builder( ).withSender( new UdpSender( ) ).build( ) )
                .build( );
    }
}
```

Trace is properly displayed in timeline and graph view:

![image](https://user-images.githubusercontent.com/1764725/76965472-a1a67000-6924-11ea-8a2f-a20b043bad89.png)

![image](https://user-images.githubusercontent.com/1764725/76965491-ac610500-6924-11ea-80d2-3d5bfa0f9f40.png)

However, DDG generation fails:

![image](https://user-images.githubusercontent.com/1764725/76965595-d74b5900-6924-11ea-9df0-9a0b51aafb18.png)

## Any open questions to address

 - Current implementation also don't support messaging scenario, ie. producer / consumer. It's not directly related to this issue so I will fill a separate one. 
 -  I would be great if the Jaeger team could document how Jaeger compute dependencies. DDG, Spark and Flink jobs seams to all behave slightly differently and several comments on this topic remained unanswered or seems to be saying the OSS and Uber version works quite differently, ex. https://github.com/jaegertracing/jaeger-ui/pull/504#issuecomment-571339764 or https://github.com/jaegertracing/jaeger-ui/issues/454#issuecomment-574412495.

**Comment by everett980** [comment]
I believe this use case was addressed by Ruben's work in [PR#557](https://github.com/jaegertracing/jaeger-ui/pull/557). Please re-open with more information if I am mistaken.

**Comment by cykl** [comment]
I just tested latest revision and confirm that #557 solves this issue. Thanks!
